code/modules/mob/living/carbon/human/life.dm

```diff
@@ -970,19 +955,18 @@
 		if(species.vision_organ)
 			vision = internal_organs_by_name[species.vision_organ]
 
-		if(!vision) // Presumably if a species has no vision organs, they see via some other means.
+		if(!species.vision_organ) // Presumably if a species has no vision organs, they see via some other means.
 			eye_blind =  0
 			blinded =    0

```

Comment:
Line 973 didn't work because vision would ALSO be null if they were missing eyes, if it was unable to pick from the internal_organs_by_name list. So it was treating that case as if the species had no eyes.